keras_retinanet/bin/train.py

```diff
@@ -109,14 +109,31 @@ def create_callbacks(model, training_model, prediction_model, validation_generat
         checkpoint = RedirectModel(checkpoint, prediction_model)
         callbacks.append(checkpoint)
 
+    if args.tensorboard_dir:
+        tensorb = keras.callbacks.TensorBoard(
+            log_dir                = args.tensorboard_dir,
+            histogram_freq         = 0,
+            batch_size             = args.batch_size,
+            write_graph            = True,
+            write_grads            = False,
+            write_images           = False,
+            embeddings_freq        = 0,
+            embeddings_layer_names = None,
+            embeddings_metadata    = None
+        )
+        callbacks.append(tensorb)
+
     if args.evaluation and validation_generator:
         if args.dataset_type == 'coco':
             from ..callbacks.coco import CocoEval
 
             # use prediction model for evaluation
             evaluation = CocoEval(validation_generator)
         else:
-            evaluation = Evaluate(validation_generator)
+            if args.tensorboard_dir:
```

Comment:
Hmm what about always initializing a `tensorboard` variable, but leave it `None` if `args.tensorboard_dir` is not set. That way this line can remain (without if):

```python
evaluation = Evaluate(validation_generator, tensorboard=tensorboard)
```